COVIDWatch iOS/Bluetooth/BluetoothPermission.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  BluetoothPermission.swift
+//  COVIDWatch iOS
+//
+//  Created by Madhava Jay on 13/4/20.
+//  Copyright © 2020 IZE. All rights reserved.
+//
+
+import Foundation
+import CoreBluetooth
+
+typealias BluetoothPermissionCallback = () -> Void
+
+class BluetoothPermission: NSObject, CBCentralManagerDelegate {
+    static let sharedInstance = BluetoothPermission()
+    override private init() {}
+    var centralManager: CBCentralManager?
+    var allowed: BluetoothPermissionCallback?
+    var denied: BluetoothPermissionCallback?
+
+    func checkPermission(_ allowed: @escaping () -> Void, _ denied: @escaping () -> Void) {
+        self.centralManager = CBCentralManager()
+        self.allowed = allowed
+        self.denied = denied
+        self.centralManager?.delegate = self
```

Comment:
Since this is a singleton class - what happens if this is called from 2 places in the app at the same time?

Reply:
Yeah, I think we can improve on this. Initially I thought the callbacks didn't exist after coming back from changing the bluetooth settings but I think that was just because the whole app was being killed. I think it can be a non singleton as long as a reference is kept.

Reply:
This is now a instance with a single result type. Its a little funky since you still kind of need to retain the instance in the local view but thats probably nicer since they will be independent and cleaned up during the view cycle.